src/status_im/navigation/core.cljs

```diff
@@ -25,7 +25,7 @@
    (fn []
      (navigation/set-default-options options/default-options)
      (reset! state/modals [])
-     (reset! state/dissmissing false)
+     (reset! state/dissmissing-count 0)
```

Comment:
Storing the dismiss state with true/false only works if we close one modal at a time. If two modals are closed simultaneously and another is opened, the state/modals behave like this: the first dismissed modal is removed, then the newly opened modal is added and immediately removed.

Reply:
Thank you. I will remove the changing-keycard-screen, so we won't need these changes.

> actually that's expected there shouldn't be an option to open 2 modals, it's wrong UI then

I'm not sure how we can fix [this PR](https://github.com/status-im/status-mobile/pull/22009) without keeping the last modal open. Should we store the old recovery phrase in the database and pass it when reopening the screen? (We probably don't need these state management changes for that PR either, as there is a delay when opening the 'Create PIN' screen. However, if we decide to keep both modals open, it might be better to keep these changes as a precaution.)

Reply:
Never mind, I will resolve that PR by passing the seed phrase during the recovery backup screen when it is stored in the database.